### PR TITLE
Added Kucoin as an exchange rate source to get more token prices

### PR DESCRIPTION
**Pull request by einsteinx2**
They seem to add tokens quickly

### CODE REVIEW

File: Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift [hunk 1]
```diff
@@ -173,6 +174,36 @@ public struct ExchangeRateParsing {
...
+        // Check for success
+        guard let success = body["success"] as? Bool, success == true else {
+            Log.error(message: "Success was false")
+            return ([], .jsonDecoding)
+        }
+        
+        // Parse the exchange rates
+        var exchangeRates = [ExchangeRate]()
+        for rateDict in data {
+            guard let coinType = rateDict["coinType"] as? String, let coinTypePair = rateDict["coinTypePair"] as? String, let lastDealPrice = rateDict["lastDealPrice"] as? Double else {
```
Comment (wolffan) [inline Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift]: we should probably do all this on its own type/class

Comment (einsteinx2) [reply Sources/BalanceServerLib/Models/Exchange Rates/ExchangeRateParsing.swift]: Yeah probably. We can talk over how to improve the architecture as we add more sources.

**Comment by einsteinx2** [comment]
Looks like the convert function is failing causing the test failure

**Comment by wolffan** [comment]
here fixed tests and conert function

**Comment by einsteinx2** [comment]
Sweet thanks!

**Comment by einsteinx2** [comment]
Merging this since you fixed the tests and I want to start collecting Kucoin prices on the backend.

**Pull request merged**